python/ray/autoscaler/_private/gcp/config.py

```diff
@@ -47,6 +48,66 @@
 # with ServiceAccounts.
 
 
+def _validate_tpu_config(node: dict):
+    """Validate the provided node with TPU support.
+
+    If the config is malformed, users will run into an error but this function
+    will raise the error at config parsing time. This only tests very simple assertions.
+
+    Raises: `ValueError` in case the input is malformed.
+
+    """
+    if "acceleratorType" in node and "acceleratorConfig" in node:
+        raise ValueError(
+            "For TPU usage, acceleratorType and acceleratorConfig "
+            "cannot both be set."
+        )
+    if "acceleratorType" in node:
+        accelerator_type = node["acceleratorType"]
+        expected_pattern = re.compile(r"^v\d+[a-zA-Z]*-\d+$")
+        if not expected_pattern.match(accelerator_type):
+            raise ValueError(
+                "`acceleratorType` should match v(generation)-(cores/chips)"
+            )
+    else:  # "acceleratorConfig" in node
+        if (
+            "type" not in node["acceleratorConfig"]
+            or "topology" not in node["acceleratorConfig"]
+        ):
+            raise ValueError("acceleratorConfig expects 'type' and 'topology'")
+        generation = node["acceleratorConfig"]["type"]
+        topology = node["acceleratorConfig"]["topology"]
+
+        generation_pattern = re.compile(r"^V\d+[a-zA-Z]*$")
+        topology_pattern = re.compile(r"^\d+x\d+(x\d+)?$")
+
+        if not generation_pattern.match(generation):
+            raise ValueError("type should match V(generation)")
+        if not topology_pattern.match(topology):
+            raise ValueError("topology should be of form axbxc or axb")
+
+
+def _get_num_tpu_chips(node: dict) -> int:
+    chips = 0
+    if "acceleratorType" in node:
+        accelerator_type = node["acceleratorType"]
+        # `acceleratorType` is typically v{chip}-{cores}
```

Comment:
```suggestion
        # `acceleratorType` is typically v{generation}-{cores}
```
Is it right? Might be slightly less confusing

Reply:
Fixed